packages/dart_frog_cli/lib/src/commands/new/new.dart

```diff
@@ -0,0 +1,157 @@
+
+  @override
+  String get name => 'new';
+
+  @override
+  final String invocation = 'dart_frog new <route|middleware> "path/to/route"';
+
+  /// Subcommand for creating a new route.
+  @visibleForTesting
+  late final DartFrogCommand newRouteCommand;
+
+  /// Subcommand for creating a new middleware.
+  @visibleForTesting
+  late final DartFrogCommand newMiddlewareCommand;
+}
+
+class _NewSubCommand extends DartFrogCommand {
+  _NewSubCommand(this.name);
+
+  @override
+  String get description => 'Create a new $name for dart_frog';
+
+  @override
+  final String name;
+
+  @override
+  NewCommand get parent => super.parent! as NewCommand;
+
+  @override
+  late final String invocation = 'dart_frog new $name "path/to/route"';
+
+  @visibleForTesting
+  @override
+  // ignore: invalid_use_of_visible_for_testing_member
+  ArgResults? get testArgResults => parent.testArgResults;
+
+  @override
+  // ignore: invalid_use_of_visible_for_testing_member
+  String? get testUsage => parent.testUsage;
+
+  String get _routePath {
+    final rest = results.rest;
+    if (rest.isEmpty) {
+      throw UsageException(
+        'Provide a route path for the new $name',
+        usageString,
+      );
+    }
+    final routeName = rest.first;
+    if (routeName.isEmpty) {
+      throw UsageException(
+        'Route path must not be empty',
+        usageString,
+      );
+    }
+
+    if (!routeName.isValidRoutePath()) {
+      throw UsageException('Route path must be a valid path segment.', '');
+    }
+
+    return routeName;
+  }
+
+  @override
+  Future<int> run() async {
+    final routePath = _routePath;
+
+    final generator = await parent._generator(dartFrogNewBundle);
+
+    final vars = <String, dynamic>{
+      'route_path': routePath,
+      'type': name,
+    };
+
+    final routesDirectory = Directory(path.join(cwd.path, 'routes'));
+    if (!routesDirectory.existsSync()) {
+      throw UsageException(
+        'No "routes" directory found in the current directory. '
+        'Make sure to run this command on a dart_frog project.',
+        usageString,
+      );
+    }
+
+    await generator.hooks.preGen(
+      vars: vars,
+      workingDirectory: cwd.path,
+      onVarsChanged: vars.addAll,
+      logger: logger,
+    );
+
+    if (!vars.containsKey('dir_path')) {
+      return ExitCode.software.code;
+    }
+
+    final generateProgress = logger.progress('Creating $name $routePath');
+    final _ = await generator.generate(
+      DirectoryGeneratorTarget(cwd),
+      vars: vars,
+    );
+
+    await generator.hooks.postGen(
+      vars: vars,
+      workingDirectory: cwd.path,
+      logger: logger,
+    );
+    generateProgress.complete();
+
+    return ExitCode.success.code;
+  }
+}
+
+extension on String {
+  bool isValidRoutePath() {
+    for (final segment in split('/').where((element) => element.isNotEmpty)) {
+      if (segment.isEmpty) {
+        return false;
```

Comment:
When is this reached? Are we not already filtering empty segments?

Reply:
Chanegdf to skipWhile, to allow '/something' but not `/something//something`